packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -417,11 +422,20 @@ export function createElement(
         type,
       );
     }
-
     if (type === 'script') {
       // Create the script via .innerHTML so its "parser-inserted" flag is
       // set to true and it does not execute
       const div = ownerDocument.createElement('div');
+      if (__DEV__ && enableTrustedTypesIntegration && !didWarnScriptTags) {
```

Comment:
Nit: please keep DEV as its own block. And nest other conditions inside of it. This makes it easy to change logic inside of DEV block without accidentally moving some out when other conditions change.